### PR TITLE
[tests] update pnpm version to 7.31.0

### PR DESCRIPTION
**Pull request by Ethan-Arrowood**
Updates our corepack pnpm version to 7.31.0 and updates the lockfile

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -10531,14 +10344,14 @@ packages:
     dev: true
 
   /fresh/0.5.2:
-    resolution: {integrity: sha512-zJ2mQYM18rEFOudeV4GShTGIQ7RbzA7ozbU9I/XBpm7kqgMywgmylMwXHxZJmkVoYkna9d2pVXVXPdYTP9ej8Q==}
+    resolution: {integrity: sha1-PYyt2Q2XZWn6g1qx+OSyOhBWBac=}
```
Comment (styfle) [inline pnpm-lock.yaml:10342]: I wonder why this one changed to sha1? 🤔

Comment (Ethan-Arrowood) [reply pnpm-lock.yaml:10342]: <picture data-single-emoji=":blob_shrug:" title=":blob_shrug:">:blob_shrug:</picture> maybe my system is messed up? Could you maybe give it a try and see if you reproduce?

**Review by EndangeredMassa — DISMISSED** [review]
Do we know exactly why the lockfile changed?

File: pnpm-lock.yaml (hunk 1 above)
Comment (styfle) [reply pnpm-lock.yaml:10342]: @Ethan-Arrowood It looks like it maintained sha512 for me (I'm using corepack)

- See #9751

**Pull request closed without merging**